lib/xMsExampleGenerator.js

```diff
@@ -0,0 +1,257 @@
+  ErrorCodes = Constants.ErrorCodes;
+
+/**
+ * @class
+ */
+class XMsExampleGenerator {
+
+  /**
+   * @constructor
+   * Initializes a new instance of the SpecResolver class.
+   * 
+   * @param {string} specPath the (remote|local) swagger spec path
+   * 
+   * @param {object} recordings the parsed spec in json format
+   * 
+   * @param {object} [options] The options object
+   * 
+   * @param {object} [options.matchApiVersion] Should relative pathes be resolved? Default: true
+   * 
+   * @param {object} [options.output] Should x-ms-examples be resolved? Default: true. 
+   * If options.shouldResolveRelativePaths is false then this option will also be false implicitly and cannot be overridden.
+   */
+  constructor(specPath, recordings, options) {
+    if (specPath === null || specPath === undefined || typeof specPath.valueOf() !== 'string' || !specPath.trim().length) {
+      throw new Error('specPath is a required property of type string and it cannot be an empty string.')
+    }
+
+    if (recordings === null || recordings === undefined || typeof recordings.valueOf() !== 'string' || !recordings.trim().length) {
+      throw new Error('recordings is a required property of type string and it cannot be an empty string.')
+    }
+
+    this.specPath = specPath;
+    this.recordings = recordings;
+    this.specDir = pathlib.dirname(this.specPath);
+    if (!options) options = {};
+    if (options.output === null || options.output === undefined) {
+      options.output = pathlib.dirname(this.specPath);
+    }
+    if (options.shouldResolveXmsExamples === null || options.shouldResolveXmsExamples === undefined) {
+      options.shouldResolveXmsExamples = true;
+    }
+    if (options.matchApiVersion === null || options.matchApiVersion === undefined) {
+      options.matchApiVersion = false;
+    }
+
+    this.options = options;
+  }
+
+  /**
+   */
+  generate() {
+    let self = this;
+    self.mkdirSync(self.options.output);
+    self.mkdirSync(self.options.output + "/examples");
+    self.mkdirSync(self.options.output + "/swagger");
+
+    let outputExamples = self.options.output + "/examples/";
+    let relativeExamplesPath = "../examples/";
+    let specName = self.specPath.split("/");
+    let outputSwagger = self.options.output + "/swagger/" + specName[specName.length - 1].split(".")[0] + ".json";
+
+    var swaggerObject = require(self.specPath);
+    var SwaggerParser = require('swagger-parser');
+    var parser = new SwaggerParser();
+
+    var example = {};
+    example['parameters'] = {};
+    example['responses'] = {};
+
+    var accErrors = {};
+    var filesArray = [];
+    self.getFileList(self.recordings, filesArray);
+
+    let recordingFiles = filesArray;
+    var example = {};
+
+    parser.parse(swaggerObject).then(function (api) {
+      console.log("API name: " + api.info.title);
+      let specApiVersion = api.info.version;
+      let paths = api.paths;
+
+      for (var recordingFile in recordingFiles) {
```

Comment:
Yup entire logic part is duplicated for now. I was just waiting to move entire login into class and then once the experience looks good we can remove the script. :)